torch/csrc/jit/passes/peephole.cpp

```diff
@@ -379,12 +379,14 @@ bool FuseAddMM(Block* block) {
             }
 
             auto* cOne = graph->insertConstant(1);
+            auto* cNone = graph->insertConstant(c10::nullopt);
             auto* addmm_node = graph->insertNode(graph->create(aten::addmm, 1));
             addmm_node->addInput(add_mat);
             addmm_node->addInput(mat1);
             addmm_node->addInput(mat2);
             addmm_node->addInput(cOne);
             addmm_node->addInput(cOne);
+            addmm_node->addInput(cNone);
```

Comment:
Here, we can pass the `dtype` from `mm` to the fused `addmm`.

There are several places in the JIT code base that we can pass through the `dtype` kwarg like this. I'm not familiar with JIT code, but I think it's something worth to do next.